core/arch/arm/plat-k3/drivers/sec_proxy.c

```diff
@@ -0,0 +1,294 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Texas Instruments K3 Secure Proxy Driver
+ *
+ * Copyright (C) 2022 Texas Instruments Incorporated - https://www.ti.com/
+ *	Manorit Chawdhry <m-chawdhry@ti.com>
+ */
+
+#include <stddef.h>
+#include <stdint.h>
+#include <platform_config.h>
+#include <trace.h>
+#include <io.h>
+#include <mm/core_mmu.h>
+#include <mm/core_memprot.h>
+#include <kernel/delay.h>
+#include <string.h>
+
+#include "sec_proxy.h"
+
+/* SEC PROXY RT THREAD STATUS */
+#define RT_THREAD_STATUS_REG            0x0
+#define RT_THREAD_STATUS_ERROR_MASK     BIT(31)
+#define RT_THREAD_STATUS_CUR_CNT_MASK   GENMASK_32(7, 0)
+
+/* SEC PROXY SCFG THREAD CTRL */
+#define SCFG_THREAD_CTRL_REG            0x1000
+#define SCFG_THREAD_CTRL_DIR_SHIFT      31
+#define SCFG_THREAD_CTRL_DIR_MASK       BIT(31)
+
+/* SECURE PROXY GENERIC HELPERS */
+enum threads {
+	SEC_PROXY_TX_THREAD,
+	SEC_PROXY_RX_THREAD,
+	SEC_PROXY_MAX_THREADS
+};
+
+#define SEC_PROXY_THREAD(base, x)       ((base) + (0x1000 * (x)))
+#define SEC_PROXY_DATA_START_OFFS       0x4
+#define SEC_PROXY_DATA_END_OFFS         0x3c
+
+#define THREAD_DIR_TX (0)
+#define THREAD_DIR_RX (1)
+
+/**
+ * struct k3_sec_proxy_thread - Description of a Secure Proxy Thread
+ * @id:		Thread ID
+ * @data:	Thread Data path region for target
+ * @scfg:	Secure Config Region for Thread
+ * @rt:		RealTime Region for Thread
+ */
+struct k3_sec_proxy_thread {
+	const char *name;
+	vaddr_t data;
+	vaddr_t scfg;
+	vaddr_t rt;
+} spts[SEC_PROXY_MAX_THREADS];
+
+/**
+ * k3_sec_proxy_verify_thread() - Verify thread status before
+ *				  sending/receiving data
+ * @dir: Direction of the thread
+ *
+ * Return: 0 if all goes well, else appropriate error message
```

Comment:
This sentence is pretty much useless, everyone knows what to expect since return type is `TEE_Result`.

Reply:
Makes sense, will remove it